vm/proxyapp/init.go

```diff
@@ -49,8 +51,17 @@ type subProcessCmd interface {
 	Wait() error
 }
 
+// Config is valid if at least cmd or rpc_server_uri specified.
```

Comment:
What will happen in both are specified?

Reply:
We'll create local app and communicate by TCP with it.

Reply:
Error handling and restarts are going to be interesting (only the connection can fail, or only the subprocess may exit).

Reply:
I would say subprocess and connection are the required resources. If something went wrong - close all the resources and allocate them again (start cmd, create connection).
But yes.... testing will be fun here... )